torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -192,6 +195,9 @@ struct GraphFuser {
         fusableDevice &= isFusableDevice(output);
       }
     }
+    for (const auto& input : node->inputs()) {
+      fusableDevice &= isFusableDevice(input);
+    }
```

Comment:
This is all getting way too complicated. Can you write out a comment describing the logic of when a FusionGroup is or isn't emitted?

Reply:
oh whoops, didn't mean to add that. it isn't necessary